src/harness/typeWriter.ts

```diff
@@ -41,7 +41,10 @@ class TypeWriterWalker {
         var lineAndCharacter = this.currentSourceFile.getLineAndCharacterOfPosition(actualPos);
         var sourceText = ts.getTextOfNodeFromSourceText(this.currentSourceFile.text, node);
 
-        var type = this.checker.getTypeAtLocation(node);
+        // Workaround to ensure we output 'C' instead of 'typeof C' for base class expressions
+        // var type = this.checker.getTypeAtLocation(node);
+        var type = node.parent && ts.isClassExtendsExpressionWithTypeArguments(node.parent) && this.checker.getTypeAtLocation(node.parent) || this.checker.getTypeAtLocation(node);

```

Comment:
In the extends clause of a class an ExpressionWithTypeArguments is considered neither a type node nor an expression node, but the contained expression is considered an expression and the contained type arguments (if any) are considered type nodes. The getTypeAtLocation function returns the base class instance type for such a node (which matches what we previously did).

In other contexts, an ExpressionWithTypeArguments and its contained nodes are all considered type nodes.